networkx/algorithms/dag.py

```diff
@@ -478,5 +484,9 @@ def dag_longest_path_length(G):
     --------
     dag_longest_path
     """
-    path_length = len(nx.dag_longest_path(G)) - 1
+    path = nx.dag_longest_path(G, weight, default_weight)
+    path_length = 0
```

Comment:
This could be more concise:
```python
edges = pairwise(path)
path_length = sum(G.edge[u][v].get(weight, default_weight) for u, v in edges)
```
Also note the use of `G.edge[u][v]` instead of `G[u][v]` to get the dictionary of edge attributes, to be consistent with other NetworkX code.